src/vec.jl

```
@@ -599,6 +599,108 @@ function Base.sum{T}(x::Vec{T})
   s[]
 end
 
+###############################################################################
+# map and friends
+import Base: map!, map
+# this should be inhertied from base
+#=
+function map(f, c)
+  out = copy(c)
+  for (idx, val) in enumerate(c)
+    out_idx = similar(val)
+    map!(f, out_idx, val)
+    out[idx] = out_idx
+  end
+
+  return out
+end
+=#
+#TODO: c should be a varargs, but that makes it ambiguous with map! defined
+# below
+function map!(f, c)
+  map!(f, c, c)
+end
+
+"""
+Applys f element-wise to src to populate dest.  If src is a ghost vector,
+then f is applied to the ghost elements as well as the local elements.
+"""
+function map!{T}(f, dest::Vec{T}, src::Vec)
+  if length(dest) != length(src)
```

I think it should throw an error if `localpart(dest) != localpart(src)` .... otherwise it won't have the same semantics as the serial `map!`